bin/pdd_renewal_report.pl

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use utf8;
+
+BEGIN {
+  die "SDRROOT environment variable not set" unless defined $ENV{'SDRROOT'};
+  use lib $ENV{'SDRROOT'} . '/crms/cgi';
+}
+
+use Getopt::Long qw(:config no_ignore_case bundling);
+
+use CRMS;
+
+use constant {
+  EXPECTED_RENEWAL_SUNSET => 68 # years
+};
+
+binmode(STDOUT, ':encoding(UTF-8)');
+my $usage = <<END;
+USAGE: $0 [-hv] [-y YEAR]
+
+Typically run a week or so into January. Reports on CRMS Core determinations with a
+now-expired renewal date for the current year that are still closed.
+
+This is a postmortem report intended to be run in January after new rights
+from the PDD rollover have taken effect. There will generally be no reason to use
+the -y flag; it is included here for consistency and in the hope it may be useful
+for future development purposes.
+
+The renDate fields that qualify are based on a current year minus 68 years scheme
+(see the magic constant EXPECTED_RENEWAL_EXPIRATION_YEARS). I will not attempt to derive
+that here except to note it relates to renewal terms in some not-particularly-obvious way;
+it is not an off-by-two derivation from 70 years but something more subtle. See Kristina
+for more information if this needs to be revisited.
+
+-h             Print this help message.
+-v             Emit verbose debugging information. May be repeated.
+-y YEAR        Use some other value for YEAR other than the current year.
+END
+
+my $help;
+my $verbose;
+my $year;
+
+Getopt::Long::Configure('bundling');
+die 'Terminating' unless GetOptions(
+           'h|?'  => \$help,
+           'v+'   => \$verbose,
+           'y:s'  => \$year);
+
+if ($help) {
+  print $usage. "\n";
+  exit 0;
+}
+
+my $crms = CRMS->new(
+  verbose  => $verbose,
+  instance => 'production'
+);
+
+$verbose = 0 unless defined $verbose;
+print "Verbosity $verbose\n" if $verbose;
+$year = $crms->GetTheYear() unless $year;
+
+my $target_year = $year - EXPECTED_RENEWAL_SUNSET;
+my $target_year_digits = substr $target_year, -2;
+print "Checking renewals for $target_year, renDate *$target_year_digits\n" if $verbose;
+
+my $jsonxs = JSON::XS->new;
+# Find Core project reviews with (any) renewal date that have not been invalidated.
+my $sql = <<'SQL';
+  SELECT r.id,rd.data FROM historicalreviews r
+  INNER JOIN exportdata e ON r.gid=e.gid
+  INNER JOIN projects p ON e.project=p.id
+  INNER JOIN reviewdata rd ON r.data=rd.id
+  WHERE r.data IS NOT NULL
+  AND r.validated=1
+  AND p.name="Core"
+  AND rd.data LIKE "%renDate%"
+  ORDER BY r.id ASC
+SQL
+
+my $ref = $crms->SelectAll($sql);
+my %seen;
+print "HTID\trenDate\tCurrent rights\n";
+foreach my $row (@$ref) {
+  my ($id, $json) = @$row;
+  next if $seen{$id};
+  my $data = $jsonxs->decode($json);
+  my $renDate = $data->{'renDate'};
+  # Narrow results down to year of interest.
```

Comment:
Could you show an example `$renDate` value in a comment? 
That way I don't have to do regexes backwards if I want to know what an expected string should look like.

Reply:
Yes, and in fact the regex itself is getting s bit of a cleanup, from `/\d+(\D\D\D)(\d\d)/` to `/\d{1,2}\D{3}(\d{2})/` so I'm not doing an unnecessary capture and the curly notation is maybe a little clearer (?). The CCE format we're looking for is "4Nov52" or "31Mar59" so I dropped those examples in.